contents/julia_101.md

```diff
@@ -423,55 +517,2108 @@ logarithm(10)
+end
+
+empty_vector
+"""
+)
+```
+
+There are some handy functions to iterate over matrices.
+
+* `eachcol`: iterates over an array column first
+
+    ```jl
+    sco(
+    """
+    first(eachcol(column_major))
+    """
+    )
+    ```
+
+* `eachrow`: iterates over an array row first
+
+    ```jl
+    sco(
+    """
+    first(eachrow(column_major))
+    """
+    )
+    ```
+
+### Pair {#sec:pair}
+
+Compared to the huge section on arrays, this section on pairs will be brief.
+**`Pair` is a data structure that holds two types**.
+How we construct a pair in Julia is using the following syntax:
+
+```jl
+sco(
+"""
+my_pair = Pair("Julia", 42)
+"""
+)
+```
+
+Alternatively, we can create a pir by specifying both values and in between we use the pair `=>` operator:
+
+```jl
+sco(
+"""
+my_pair = "Julia" => 42
+"""
+)
+```
+
+The elements are stored in the fields `first` and `second`.
+
+```jl
+scob(
+"""
+my_pair.first
+"""
+)
+```
+
+```jl
+scob(
+"""
+my_pair.second
+"""
+)
+```
+
+Pairs will be used a lot in data manipulation and data visualization since both [`DataFrames.jl`](dataframes.html) and [`Plots.jl`](plots.html) main functions depends on `Pair` as type arguments.
+
+### Dict {#sec:dict}
+
+If you understood what a `Pair` is, then `Dict` won't be a problem.
+**`Dict` in Julia is just a "hash table" with pairs of `key` and `value`**.
+`key`s and `value`s can be of any type, but generally you'll see `key`s as strings.
+
+There are two ways to construct `Dict`s in Julia.
+The first is using the **default constructor `Dict` and passing a vector of tuples composed of `(key, value)`**:
+
+```jl
+sco(
+"""
+my_dict = Dict([("one", 1), ("two", 2)])
+"""
+)
+```
+
+We *prefer* the second way of constructing `Dict`s.
+It offers a much elegant and expressive syntax.
+You use the same **default constructor `Dict`**, but now you pass **`pair`s of `key` and `value`**:
+
+
+```jl
+sco(
+"""
+my_dict = Dict("one" => 1, "two" => 2)
+"""
+)
+```
+
+You can retrieve a `Dict`s `value` by indexing it by the corresponding `key`:
+
+```jl
+scob(
+"""
+my_dict["one"]
+"""
+)
+```
+
+Similarly, to add a new entry you index the `Dict` by the desired `key` and assign a `value` with the assignment `=` operator:
+
+```jl
+scob(
+"""
+my_dict["three"] = 3
+"""
+)
+```
+
+If you want to check if a `Dict` has a certain `key` you can use the `haskey` function:
+
+```jl
+scob(
+"""
+haskey(my_dict, "two")
+"""
+)
+```
+
+To delete a `key` you can use either the `delete!` function:
+
+```jl
+sco(
+"""
+delete!(my_dict, "three")
+"""
+)
+```
+
+Or to delete a `key` while retuning its `value` you can use the `pop!` function:
+
+```jl
+scob(
+"""
+popped_value = pop!(my_dict, "two")
+"""
+)
+```
+
+Now our `my_dict` has only one `key`:
+
+```jl
+scob(
+"""
+length(my_dict)
+"""
+)
+```
+
+```jl
+sco(
+"""
+my_dict
+"""
+)
+```
+
+`Dict`s are also used in data manipulations by [`DataFrames.jl`](dataframes.html) and data visualization by [`Plots.jl`](plots.html).
```

Comment:
Maybe, also add the `Dict(zip(A, B))` where `A` and `B` are vectors, that is, iterable. It's a super useful constructor if you quickly want to have a mapping from `A` to `B`.

Reply:
Ok done in 8867c5a8ff5e147361ba4b5d6dd36f85e4375fbe